checks.d/snmp.py

```diff
@@ -123,49 +139,89 @@ def check_table(self, instance, oids, lookup_names):
-                    metric =  object[1]
-                    indexes = object[2]
-                    results[metric][indexes] = value
+        while first_oid < len(oids):
+
+            # Start with snmpget command
+            snmp_command = self.cmd_generator.getCmd
+            error_indication, error_status, error_index, var_binds = snmp_command(
+                auth_data,
+                transport_target,
+                *(oids[first_oid:first_oid + OID_BATCH_SIZE]),
+                lookupValues=lookup_names,
+                lookupNames=lookup_names)
+
+            # Raise on error_indication
+            self.raise_on_error_indication(error_indication, instance)
+
+            # Continue on error_status
+            if error_status:
+                message = "{0} for instance {1}".format(error_status.prettyPrint(),
+                                                        instance["ip_address"])
+                instance["service_check_error"] = message
+                self.log.warning(message)
+                continue
+
+            missing_results = []
+            complete_results = []
+
+            for var in var_binds:
+                result_oid, value = var
+                if reply_invalid(value):
+                    oid_tuple = result_oid.asTuple()
+                    oid = ".".join([str(i) for i in oid_tuple])
+                    missing_results.append(oid)
                 else:
-                    oid = result_oid.asTuple()
-                    matching = ".".join([str(i) for i in oid])
-                    results[matching] = value
+                    complete_results.append(var)
+
+            if missing_results:
+                # If we didn't catch the metric using snmpget, try snmpnext
+                snmp_command = self.cmd_generator.nextCmd
+                error_indication, error_status, error_index, var_binds_table = snmp_command(
+                    auth_data,
+                    transport_target,
+                    *missing_results,
+                    lookupValues=lookup_names,
+                    lookupNames=lookup_names)
+
+                # Raise on error_indication
+                self.raise_on_error_indication(error_indication, instance)
+
+                # Continue on error_status
+                if error_status:
+                    message = "{0} for instance {1}".format(error_status.prettyPrint(),
+                                                            instance["ip_address"])
+                    instance["service_check_error"] = message
+                    self.log.warning(message)
+                    continue
+                for table_row in var_binds_table:
+                    complete_results.extend(table_row)
+
+            all_binds.extend(complete_results)
+            first_oid = first_oid + OID_BATCH_SIZE
+
+        for result_oid, value in all_binds:
+            if lookup_names:
+                object = result_oid.getMibSymbol()
```

Comment:
Avoid the variable name `object` since it already means something in Python, and it doesn't indicate what is inside.